thorlcr/activities/indexwrite/thindexwriteslave.cpp

```diff
@@ -699,6 +709,19 @@ class IndexWriteSlaveActivity : public ProcessSlaveActivity, public ILookAheadSt
     virtual void gatherActiveStats(CRuntimeStatisticCollection &activeStats) const
     {
         PARENT::gatherActiveStats(activeStats);
+        {
+            CriticalBlock b(builderFileCS);
+            if (builderIFileIO)
+            {
+                mergeStats(activeStats, builderIFileIO, diskWriteRemoteStatistics);
```

Comment:
from JIRA description:
>The IKeyBuilder should expose the stats, so that gatherActiveStats can update the serialized stats. covering both scenarios.

I think all these stats should be coming from the builder (including the io stats), so that it a single call:
```
mergeStats(activeStats, builder, indexWriteActivityStatistics);
```

There's other stats that are currently be manually pulled out of the builder and serialize to the manager in a non-standardized stats way at the moment, that could also be directly gather from the IKeyBuilder